### PR TITLE
show proper error when using a sysroot without MIR

### PR DESCRIPTION
**Pull request by RalfJung**
Fixes https://github.com/rust-lang/miri/issues/1834

### CODE REVIEW

File: src/diagnostics.rs [hunk 1]
```diff
@@ -132,8 +132,6 @@ pub fn report_error<'tcx, 'mir>(
             };
             #[rustfmt::skip]
             let helps = match e.kind() {
-                Unsupported(UnsupportedOpInfo::NoMirFor(..)) =>
```
Comment (RalfJung) [inline src/diagnostics.rs:135]: This error is never raised by Miri anyway, and I will try to remove this error variant from rustc.

**Comment by RalfJung** [comment]
@oli-obk I should probably have waited instead of landing this myself... sorry for that. I can rework this to a different approach if you'd prefer that. (The alternative is to have a query that returns `Option<mir::Body>`, instead of ICEing when there is no MIR body.)

**Comment by oli-obk** [comment]
ICEing is fine by me, I see no use case for handling it gently right now

**Comment by RalfJung** [comment]
Well the usecase would be that we'd not have to use this hack of checking a sentinel function; instead we'd *try* to load the MIR on each function call and then throw a proper interpreter error when it is missing.

**Pull request merged**